### PR TITLE
mcs: fix unnecessary PDRedirectorHeader

### LINKED ISSUE #7533: mcs: a lot of `redirect but server is not leader` in test
### What did you do?

run placement rules test but failed
https://tcms.pingcap.net/dashboard/executions/plan/5131009

### What did you expect to see?

success

### What did you see instead?

image

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7533

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/apiutil/serverapi/middleware.go [hunk 1]
```diff
@@ -182,14 +182,16 @@ func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http
 		return
 	}
 
-	// Prevent more than one redirection.
-	if name := r.Header.Get(apiutil.PDRedirectorHeader); len(name) != 0 {
-		log.Error("redirect but server is not leader", zap.String("from", name), zap.String("server", h.s.Name()), errs.ZapError(errs.ErrRedirect))
-		http.Error(w, errs.ErrRedirectToNotLeader.FastGenByArgs().Error(), http.StatusInternalServerError)
-		return
+	// Prevent more than one redirection among PD/API servers.
+	if !redirectToMicroService {
```
Comment (rleungx) [inline pkg/utils/apiutil/serverapi/middleware.go:186]: Can we merge it with line 207?

File: tests/pdctl/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -691,47 +690,3 @@ func mightExec(re *require.Assertions, cmd *cobra.Command, args []string, v inte
 	}
 	json.Unmarshal(output, v)
 }
-
-func TestForwardSchedulerRequest(t *testing.T) {
```
Comment (rleungx) [inline tests/pdctl/scheduler/scheduler_test.go:695]: Why remove it?

Comment (lhy1024) [reply tests/pdctl/scheduler/scheduler_test.go:695]: it is duplicated with TestScheduler, when TestScheduler support to run in two modes.

File: tests/integrations/mcs/scheduling/api_test.go [hunk 3]
```diff
@@ -499,3 +498,41 @@ func (suite *apiTestSuite) checkAdminRegionCacheForward(cluster *tests.TestClust
 	re.Equal(0, schedulingServer.GetCluster().GetRegionCount([]byte{}, []byte{}))
 	re.Equal(0, apiServer.GetRaftCluster().GetRegionCount([]byte{}, []byte{}).Count)
 }
+
+func (suite *apiTestSuite) TestFollowerForward() {
+	suite.env.RunTestInAPIMode(suite.checkFollowerForward)
```
Comment (rleungx) [inline tests/integrations/mcs/scheduling/api_test.go:503]: Shall we also test it in pd mode?

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**